extensions/ql-vscode/test/vscode-tests/no-workspace/databases/github-repository-finder.test.ts

```diff
@@ -0,0 +1,251 @@
+import { Extension, extensions, Uri } from "vscode";
+import * as workspaceFolders from "../../../../src/common/vscode/workspace-folders";
+import {
+  GitExtension,
+  API as GitExtensionAPI,
+} from "../../../../src/common/vscode/extension/git";
+import { mockedObject } from "../../utils/mocking.helpers";
+import { findGitHubRepositoryForWorkspace } from "../../../../src/databases/github-repository-finder";
+import { ValueResult } from "../../../../src/common/value-result";
+
+describe("findGitHubRepositoryForWorkspace", () => {
+  let mockGitExtensionAPI: GitExtensionAPI;
+
+  let getOnDiskWorkspaceFolderObjectsSpy: jest.SpiedFunction<
+    typeof workspaceFolders.getOnDiskWorkspaceFoldersObjects
+  >;
+  let getExtensionSpy: jest.SpiedFunction<typeof extensions.getExtension>;
+  const getAPISpy: jest.MockedFunction<GitExtension["getAPI"]> = jest.fn();
+
+  const repositories = [
+    {
+      rootUri: Uri.file("a/b/c"),
+      state: {
+        HEAD: {
+          name: "main",
+          upstream: {
+            name: "origin",
```

Comment:
It is slightly confusing that a lot of this is actually not needed for the test to work. We could get rid of the whole HEAD part and it'll just take the first remote.

It might be ok, unless you want to actually test the heuristics, in which case we probably want a few different repos to get the different cases.

Reply:
Good point, I've added some extra tests to cover the heuristics and changed this specific state to not have an `origin` upstream.